pkg/platform/model/buildplanner.go

```diff
@@ -328,3 +340,31 @@ func (bp *BuildPlanner) GetBuildExpression(owner, project, commitID string) (*bu
 
 	return expression, nil
 }
+
+func FormatBuildPlanError(bperr *bpModel.BuildPlannerError) error {
+	var err error = bperr
+	// Append last five lines to error message
+	offset := 0
+	numLines := len(bperr.ValidationErrors)
+	if numLines > 5 {
+		offset = numLines - 5
+	}
+
+	errorLines := strings.Join(bperr.ValidationErrors[offset:], "\n")
+	// Crop at 500 characters to reduce noisy output further
+	if len(errorLines) > 500 {
+		offset = len(errorLines) - 499
+		errorLines = fmt.Sprintf("…%s", errorLines[offset:])
+	}
```

Comment:
How big do these messages get that this was necessary? :o

Reply:
This is repurposed legacy code that originally processed solver errors. I guess we encountered some fairly large ones in the past :P